beekeeper-scheduler-apiary/src/main/java/com/expediagroup/beekeeper/scheduler/apiary/generator/utils/CleanupDelayExtractor.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (C) 2019-2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.beekeeper.scheduler.apiary.generator.utils;
+
+import java.time.Duration;
+import java.time.format.DateTimeParseException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.expedia.apiary.extensions.receiver.common.event.ListenerEvent;
+
+public class CleanupDelayExtractor {
+
+  private static final Logger log = LoggerFactory.getLogger(CleanupDelayExtractor.class);
+  private final String propertyKey;
+  private final String defaultValue;
+
+  public CleanupDelayExtractor(String propertyKey, String defaultValue) {
+    this.propertyKey = propertyKey;
+    this.defaultValue = defaultValue;
+  }
+
+  public Duration extractCleanupDelay(ListenerEvent listenerEvent) {
+    String tableCleanupDelay = listenerEvent.getTableParameters().getOrDefault(propertyKey, defaultValue);
+
+    try {
+      return Duration.parse(tableCleanupDelay);
+    } catch (DateTimeParseException e) {
+      log.error("Text '{}' cannot be parsed to a Duration for table '{}.{}'. Using default setting.",
+          tableCleanupDelay, listenerEvent.getDbName(), listenerEvent.getTableName());
+      return Duration.parse(defaultValue);
```

Comment:
I think we should parse this in the constructor so that we don't have to keep on doing it, and so that we can catch any misconfiguration when Beekeeper starts up and not when paths are scheduled.

Reply:
I'm generally a fan for failing as fast as possible too.

Reply:
@vedantchokshi